### PR TITLE
Expose option to log internal exceptions

### PR DESCRIPTION
**Pull request by sbomer**
Also enable it for ILLink.Tasks, which will result in resolution
errors being logged as warnings during publish.

@erozenfeld PTAL.

### CODE REVIEW

File: linker/Linker/Driver.cs [hunk 1]
```diff
@@ -346,6 +351,7 @@ static void Usage (string msg)
 			Console.WriteLine ("   --about             About the {0}", _linker);
 			Console.WriteLine ("   --version           Print the version number of the {0}", _linker);
 			Console.WriteLine ("   --skip-unresolved   Ignore unresolved types, methods, and assemblies (true or false)");
+			Console.WriteLine ("   --log-warnings      Log warnings for internal exceptions");
```
Comment (erozenfeld) [inline linker/Linker/Driver.cs]: If I'm reading the code correctly turning on this option will enable all LogMessage output, not just warnings. I think both the option name and the description are misleading. Maybe --log-messages or --output-log?

Comment (sbomer) [reply linker/Linker/Driver.cs]: They are a bit misleading. I changed them to LogMessages and --log-messages.

File: linker/Linker/Driver.cs [hunk 2]
```diff
@@ -346,6 +350,7 @@ static void Usage (string msg)
 			Console.WriteLine ("   --about             About the {0}", _linker);
 			Console.WriteLine ("   --version           Print the version number of the {0}", _linker);
 			Console.WriteLine ("   --skip-unresolved   Ignore unresolved types, methods, and assemblies (true or false)");
+			Console.WriteLine ("   --log-messages      Log messages indicating progress and warnings");
```
Comment (marek-safar) [inline linker/Linker/Driver.cs]: I'd prefer `--verbose`

File: linker/Linker/Driver.cs [hunk 3]
```diff
@@ -112,8 +112,12 @@ public void Run (ILogger customLogger = null)
 							continue;
 						}
 
-						if (token == "--dependencies-file")
-						{
+                                                if (token == "--log-messages") {
```
Comment (marek-safar) [inline linker/Linker/Driver.cs]: Indentation is quite off

File: linker/Linker/Driver.cs (hunk 2 above)
Comment (sbomer) [reply linker/Linker/Driver.cs]: Changed.

File: linker/Linker/Driver.cs (hunk 3 above)
Comment (sbomer) [reply linker/Linker/Driver.cs]: Fixed.

**Pull request merged**